gammapy/cube/fit.py

```diff
@@ -867,11 +868,20 @@ def to_image(self, spectrum=None, keepdims=True):
         dataset : `MapDataset`
             Map dataset containing images.
         """
+        self.counts.data[~self.mask_safe.data] = 0
```

Comment:
Note that this modifies the data in place and leaves the parent `MapDataset` in a modified state. This we should avoid and make a copy of the data instead. This can be achieved e.g. by just multiplying the mask as well:

```python3
from gammapy.maps import WcsNDMap
import numpy as np

m = WcsNDMap.create(width=(8, 5))
m.data += 1

mask = np.ones(m.geom.data_shape)
mask[10:20] = 0

m_masked = m * mask

m_masked.plot()
```

We could also add a `weights` argument to `.sum_over_axis()`.